### DISCUSSION TITLE
Will the relay return all transactions in the future?

### DISCUSSION THREAD

**Discussion by tom-hbar** [hiero-ledger/hiero-json-rpc-relay#567] [Q&A] [answered]
After some investigation, it seems that `eth_getBlockByNumber` only returns transactions that are of type `CONTRACTCALL` or `ETHEREUMTRANSACTION`. The transactions array is empty if no transactions of those types are in the block, despite there being >0 transactions in the block.

Additionally, `eth_getTransactionByHash` only returns information about transactions of type `CONTRACTCALL` or `ETHEREUMTRANSACTION`. It also looks like it only returns `CONTRACTCALL` transactions that were submitted via the relay – HAPI contract calls via the SDK return null.

Will the relay return all transactions (for example, `CRYPTOTRANSFER`) for these methods at some point in the future? If so, is there a timeline for implementation?

**Comment by Nana-EC** [comment, marked as answer]
Hi @tom-hbar 
Thanks for the questions. Apologies for the late reply we missed this. 

The Ethereum JSON RPC Spec only applies to EVM type transactions. This means transactions of type `CONTRACTCREATE`,`CONTRACTCALL` and  `ETHEREUMTRANSACTION`.
Other HAPI transactions will not be supported with the exception of HTS tokens transfers.

This may have been a bug at the time, lots of improvements since then.
Notably, we've added support for synthetic transactions that capture the event logs of HTS transfers similar to ERC20 and ERC721 events.

These should be sufficient for EVM developers as it matches behaviour of other EVM chains.
If you'd like details of non EVM executions transaction the Mirror Node API would be the appropriate source.
 
Let me know if you believe there's a gap.
Thanks again

**Comment by Nana-EC** [comment]
Closing off.
The Ethereum JSON RPC API is focused on transaction that go through the EVM.
They don't have support for Hedera transactions outside of the EVM.
For those you call query from the Mirror Node API https://docs.hedera.com/hedera/sdks-and-apis/rest-api
